pkg/runtime/typescript.dockerfile

```diff
@@ -14,36 +15,31 @@ COPY package.json *.lock *-lock.json /
 
 RUN yarn import || echo ""
 
-RUN set -ex && \
-    yarn install --production --frozen-lockfile --cache-folder /tmp/.cache && \
-    rm -rf /tmp/.cache
+RUN --mount=type=cache,target=/tmp/.yarn_cache \
+    set -ex && \
+    yarn install --production --prefer-offline --frozen-lockfile --cache-folder /tmp/.yarn_cache
 
 RUN test -f tsconfig.json || echo "{\"compilerOptions\":{\"esModuleInterop\":true,\"target\":\"es2015\",\"moduleResolution\":\"node\"}}" > tsconfig.json
 
 COPY . .
 
 # make prisma external to bundle - https://github.com/prisma/prisma/issues/16901#issuecomment-1362940774 \
 # TODO: remove when custom dockerfile support is available
-RUN \
-  --mount=type=cache,target=/tmp/ncc-cache \
+RUN --mount=type=cache,target=/tmp/ncc-cache \
   ncc build ${HANDLER} -o lib/ -e .prisma/client -e @prisma/client -t
 
 FROM node:alpine as final
 
-WORKDIR /usr/app
-
 RUN apk update && \
     apk add --no-cache ca-certificates && \
     update-ca-certificates
 
-COPY package.json *.lock *-lock.json ./
-
-RUN set -ex && \
-    yarn install --production --frozen-lockfile --cache-folder /tmp/.cache && \
-    rm -rf /tmp/.cache
+WORKDIR /usr/app
 
 COPY . .
 
+COPY --from=build node_modules/ ./node_modules/
```

Comment:
Need to verify this change (the copy of the json/lock files above are actually being copied into the container root